src/framework/audio/internal/platform/win/wasapiaudioclient.cpp

```diff
@@ -536,11 +571,24 @@ void WasapiAudioClient::getSamples(uint32_t framesAvailable)
     uint8_t* data;
 
     uint32_t actualFramesToRead = framesAvailable;
-    uint32_t actualBytesToRead = actualFramesToRead * m_mixFormat->nBlockAlign;
+
+    // MuseScore assumes only 2 audio channels, while client may have more; we'll fill the first two
+    uint32_t muFrameSize = 2 * m_mixFormat->wBitsPerSample / 8;
+    uint32_t clientFrameSize = m_mixFormat->nBlockAlign;
 
     check_hresult(m_audioRenderClient->GetBuffer(actualFramesToRead, &data));
-    if (actualBytesToRead > 0) {
-        m_sampleRequestCallback(nullptr, data, actualBytesToRead);
+    if (actualFramesToRead > 0) {
+        if (clientFrameSize > muFrameSize) {
+            std::vector<uint8_t> tempBuffer(actualFramesToRead* muFrameSize);
+            m_sampleRequestCallback(nullptr, tempBuffer.data(), tempBuffer.size());
+
+            for (uint32_t i = 0; i < actualFramesToRead; ++i) {
+                std::memcpy(data + i * clientFrameSize, tempBuffer.data() + i * muFrameSize, muFrameSize);
+                std::memset(data + i * clientFrameSize + muFrameSize, 0, clientFrameSize - muFrameSize);
+            }

```

Comment:
This is the ugly part...

Reply:
Not that ugly if there's a comment in the code referencing this PR. It makes sense for now for a quick fix for the crashes to just create a bigger sample buffer with blank data for all the extra channels not created by MuseScore.